pkg/controller/jobs/pod/pod_controller_test.go

```diff
@@ -0,0 +1,489 @@
+
+			for i := range tc.workloads {
+				kcBuilder = kcBuilder.WithStatusSubresource(&tc.workloads[i])
+			}
+
+			kClient := kcBuilder.Build()
+			for i := range tc.workloads {
+				if err := ctrl.SetControllerReference(&tc.pod, &tc.workloads[i], kClient.Scheme()); err != nil {
+					t.Fatalf("Could not setup owner reference in Workloads: %v", err)
+				}
+				if err := kClient.Create(ctx, &tc.workloads[i]); err != nil {
+					t.Fatalf("Could not create workload: %v", err)
+				}
+			}
+			recorder := record.NewBroadcaster().NewRecorder(kClient.Scheme(), corev1.EventSource{Component: "test"})
+			reconciler := NewReconciler(kClient, recorder)
+
+			podKey := client.ObjectKeyFromObject(&tc.pod)
+			_, err := reconciler.Reconcile(ctx, reconcile.Request{
+				NamespacedName: podKey,
+			})
+			if diff := cmp.Diff(tc.wantErr, err, cmpopts.EquateErrors()); diff != "" {
+				t.Errorf("Reconcile returned error (-want,+got):\n%s", diff)
+			}
+
+			var gotPod corev1.Pod
+			if err := kClient.Get(ctx, podKey, &gotPod); err != nil {
+				t.Fatalf("Could not get Pod after reconcile: %v", err)
+			}
+			if diff := cmp.Diff(tc.wantPod, gotPod, podCmpOpts...); diff != "" {
+				t.Errorf("Pod after reconcile (-want,+got):\n%s", diff)
+			}
+			var gotWorkloads kueue.WorkloadList
+			if err := kClient.List(ctx, &gotWorkloads); err != nil {
+				t.Fatalf("Could not get Workloads after reconcile: %v", err)
+			}
+			if diff := cmp.Diff(tc.wantWorkloads, gotWorkloads.Items, tc.workloadCmpOpts...); diff != "" {
+				t.Errorf("Workloads after reconcile (-want,+got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestIsPodOwnerManagedByQueue(t *testing.T) {
+	testCases := map[string]struct {
+		ownerReference metav1.OwnerReference
+		wantRes        bool
+	}{
+		"batch/v1/Job": {
+			ownerReference: metav1.OwnerReference{
+				APIVersion: "batch/v1",
+				Controller: ptr.To(true),
+				Kind:       "Job",
+			},
+			wantRes: true,
+		},
+		"apps/v1/ReplicaSet": {
+			ownerReference: metav1.OwnerReference{
+				APIVersion: "apps/v1",
+				Controller: ptr.To(true),
+				Kind:       "ReplicaSet",
+			},
+			wantRes: false,
+		},
+		"ray.io/v1alpha1/RayCluster": {
+			ownerReference: metav1.OwnerReference{
+				APIVersion: "ray.io/v1alpha1",
+				Controller: ptr.To(true),
+				Kind:       "RayCluster",
+			},
+			wantRes: true,
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			pod := testingutil.MakePod("pod", "ns").
+				UID("test-uid").
+				Request(corev1.ResourceCPU, "1").
+				Image("", nil).
+				Obj()
+
+			pod.OwnerReferences = append(pod.OwnerReferences, tc.ownerReference)
+
+			if diff := cmp.Diff(tc.wantRes, IsPodOwnerManagedByQueue((*Pod)(pod))); diff != "" {
+				t.Errorf("Unexpected 'IsPodOwnerManagedByQueue' result (-want,+got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestGetWorkloadNameForPod(t *testing.T) {
+	wlName := GetWorkloadNameForPod("unit-test")
+
+	if diff := cmp.Diff("pod-unit-test-7bb47", wlName); diff != "" {
```

Comment:
```suggestion
	if "pod-unit-test-7bb47" != wlName {
```

Same reason for the above.